### PR TITLE
Introduce ranged sequence request

### PR DESCRIPTION
**Pull request by zvirja**
As promised, created a new kind of request - `RangedSequenceRequest`. It's purpose is to activate a sequence withing the specified boundaries.

Currently I'm simply picking a random value withing a range. Later we might change the strategy if really needed.

@moodmosaic Please take a look 😉

### CODE REVIEW

File: Src/AutoFixture/EnumerableExtensions.cs [hunk 1]
```diff
@@ -20,17 +27,17 @@ internal static int IndexOf<T>(this IEnumerable<T> items, T item)
             return -1;
         }
 
-        internal static IEnumerable<T> Insert<T>(this IEnumerable<T> items, int index, T item)
+        public static IEnumerable<T> Insert<T>(this IEnumerable<T> items, int index, T item)
```
Comment (moodmosaic) [inline Src/AutoFixture/EnumerableExtensions.cs]: You sure you wan't this public?

File: Src/AutoFixture/EnumerableExtensions.cs [hunk 2]
```diff
@@ -20,17 +27,17 @@ internal static int IndexOf<T>(this IEnumerable<T> items, T item)
...
         {
             return items.Take(index).Concat(new[] { item }).Concat(items.Skip(index));
         }
 
-        internal static IEnumerable<T> RemoveAt<T>(this IEnumerable<T> items, int index)
+        public static IEnumerable<T> RemoveAt<T>(this IEnumerable<T> items, int index)
```
Comment (moodmosaic) [inline Src/AutoFixture/EnumerableExtensions.cs]: You sure you wan't this public?

File: Src/AutoFixture/EnumerableExtensions.cs [hunk 3]
```diff
@@ -20,17 +27,17 @@ internal static int IndexOf<T>(this IEnumerable<T> items, T item)
...
         {
             return items.Take(index).Concat(items.Skip(index + 1));
         }
 
-        internal static IEnumerable<T> SetItem<T>(this IEnumerable<T> items, int index, T item)
+        public static IEnumerable<T> SetItem<T>(this IEnumerable<T> items, int index, T item)
```
Comment (moodmosaic) [inline Src/AutoFixture/EnumerableExtensions.cs]: You sure you wan't this public?

File: Src/AutoFixture/EnumerableExtensions.cs [hunk 4]
```diff
@@ -41,5 +48,19 @@ internal static IEnumerable<T> SetItem<T>(this IEnumerable<T> items, int index,
 
             return a.Take(index).Concat(new[] { item }).Concat(a.Skip(index + 1));
         }
+
+        public static object ToTypedArray(this IEnumerable items, Type elementType)
```
Comment (moodmosaic) [inline Src/AutoFixture/EnumerableExtensions.cs]: Perhaps it can be internal?

File: Src/AutoFixture/EnumerableExtensions.cs (hunk 1 above)
Comment (zvirja) [reply Src/AutoFixture/EnumerableExtensions.cs]: Notice, the `EnumerableExtensions` class is internal, so all the methods are still internal, even though they are marked as `public`.

File: Src/AutoFixture/EnumerableExtensions.cs (hunk 2 above)
Comment (zvirja) [reply Src/AutoFixture/EnumerableExtensions.cs]: Notice, the `EnumerableExtensions` class is internal, so all the methods are still internal, even though they are marked as `public`.

File: Src/AutoFixture/EnumerableExtensions.cs (hunk 3 above)
Comment (zvirja) [reply Src/AutoFixture/EnumerableExtensions.cs]: Notice, the `EnumerableExtensions` class is internal, so all the methods are still internal, even though they are marked as `public`.

File: Src/AutoFixture/EnumerableExtensions.cs (hunk 4 above)
Comment (zvirja) [reply Src/AutoFixture/EnumerableExtensions.cs]: Notice, the `EnumerableExtensions` class is internal, so all the methods are still internal, even though they are marked as `public`. This particular method is indeed internal, I don't want to make it public.

**Comment by zvirja** [comment]
@moodmosaic From the design I have in my head, I definitely see value in all the requests/relays we currently have. All them cover the different areas:
- `MinAndMaxLengthAttributeRelay` - it knows about the data annotations attributes and how they should be applied.
- `RangedSequenceRequest/RangedSequenceRelay` - an abstraction for the AutoFixture core to create a sequence with the length within the specified range. It decides the exact length.
- `FiniteSequenceRequest` - request for the sequence of exact length.

I don't see the types we could strip. If you see some other design - please share your vision 👍

**Pull request merged**